src/storage-controller/src/lib.rs

```diff
@@ -1911,9 +1911,42 @@ where
         );
         self.set_hold_policies(drop_policy);
 
+        // Record the drop status for all sink drops.
+        //
+        // We also delete the items' statistics objects.
+        //
+        // The locks are held for a short time, only while we do some hash map removals.
```

Comment:
Nit: They are all btree maps, no? :)

Reply:
Very true, I just copied this comment over ... 🙈